datasets/saudinewsnet/saudinewsnet.py

```diff
@@ -0,0 +1,98 @@
+
+from __future__ import absolute_import, division, print_function
+
+import json
+import os
+
+import datasets
+
+
+_CITATION = """\
+@misc{hagrima2015, author = "M. Alhagri", title = "Saudi Newspapers Arabic Corpus \
+(SaudiNewsNet)", year = 2015, url = "http://github.com/ParallelMazen/SaudiNewsNet" }
+"""
+
+
+_DESCRIPTION = """The dataset contains a set of 2,972 Arabic newspaper articles alongwith metadata, \
+extracted from various online Saudi newspapers and written in MSA."""
+
+
+_HOMEPAGE = "https://github.com/parallelfold/SaudiNewsNet"
+
+
+_LICENSE = "Creative Commons Attribution-NonCommercial-ShareAlike 4.0 International License."
+
+
+_URL = "https://github.com/parallelfold/SaudiNewsNet/raw/master/dataset/2015-07-21.zip"
+
+
+class SaudiNewsNet(datasets.GeneratorBasedBuilder):
+    """a set of 2972 Arabic newspaper articles along with metadata"""
+
+    def _info(self):
+        return datasets.DatasetInfo(
+            description=_DESCRIPTION,
+            features=datasets.Features(
+                {
+                    "source": datasets.Value(
+                        "string"
+                    ),  # A string identifief of the newspaper from which the article was extracted.
+                    "url": datasets.Value("string"),  # The full URL from which the article was extracted.
+                    "date_extracted": datasets.Value(
+                        "string"
+                    ),  # The timestamp of the date on which the article was extracted.
+                    "title": datasets.Value("string"),  # The title of the article. Can be empty.
+                    "author": datasets.Value("string"),  # The author of the article. Can be empty.
+                    "content": datasets.Value("string"),  # The content of the article.
+                }
+            ),
+            homepage=_HOMEPAGE,
+            citation=_CITATION,
+            supervised_keys=None,
+        )
+
+    def _split_generators(self, dl_manager):
+        """Returns SplitGenerators."""
+        datadir = dl_manager.download_and_extract(_URL)
+        return [
+            datasets.SplitGenerator(
+                name=datasets.Split.TRAIN,
+                gen_kwargs={"filepath": os.path.join(datadir, "2015-07-21.json"), "split": "train"},
+            )
+        ]
+
+    def _generate_examples(self, filepath, split):
+        """Generates examples"""
+        with open(filepath, encoding="utf-8") as f:
+            articles = json.load(f)
+            for _id, article in enumerate(articles):
+                title = article.get("title", "").strip()
+                source = article.get("source", "").strip()
+                dt = article.get("date_extracted", "").strip()
+                link = article.get("url", "").strip()
+                author = article.get("author", "").strip()
+                content = article.get("content", "").strip()
```

Comment:
Are there missing values in the dataset ?

If yes, in which fields ?

For those fields you can you mention in the dataset card that the missing values are replaced with empty strings, in the section `Data Fields`.

If not, then please remove the `get` of the fields that don't have missing values.